pkg/http/transport_test.go

```diff
@@ -28,3 +28,26 @@ func TestTransportAddsHeaders(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, resp.Request.Header.Get(testHeader), testValue)
 }
+
+func TestTransportOnlyAddsHeaderIfMissing(t *testing.T) {
+	// Setup mock http server
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	}))
+	defer server.Close()
+
+	const testHeader = "X-Test-Header"
```

Comment:
is `const` required for the test? it's fine, just uncommon to see in a function body

Reply:
I suppose I don't expect it to change jajaja